x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.js

```diff
@@ -27,57 +23,97 @@ export class HeadlessChromiumDriverFactory {
 
   type = 'chromium';
 
-  create({ viewport, browserTimezone }) {
-    return Rx.Observable.create(async observer => {
-      const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
-      const chromiumArgs = args({
+  async create({ viewport, browserTimezone }) {
+    const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
+    const chromiumArgs = args({
+      userDataDir,
+      viewport,
+      verboseLogging: this.logger.isVerbose,
+      disableSandbox: this.browserConfig.disableSandbox,
+      proxyConfig: this.browserConfig.proxy,
+    });
+
+    let chromium;
+    let page;
+
+    try {
+      chromium = await puppeteer.launch({
         userDataDir,
-        viewport,
-        verboseLogging: this.logger.isVerbose,
-        disableSandbox: this.browserConfig.disableSandbox,
-        proxyConfig: this.browserConfig.proxy,
+        executablePath: this.binaryPath,
+        ignoreHTTPSErrors: true,
+        args: chromiumArgs,
+        env: {
+          TZ: browserTimezone
+        },
       });
 
-      let chromium;
-      let page;
-      try {
-        chromium = await puppeteer.launch({
-          userDataDir,
-          executablePath: this.binaryPath,
-          ignoreHTTPSErrors: true,
-          args: chromiumArgs,
-          env: {
-            TZ: browserTimezone
-          },
-        });
-
-        page = await chromium.newPage();
-      } catch (err) {
-        observer.error(new Error(`Caught error spawning Chromium`));
-        return;
-      }
 
-      safeChildProcess({
-        async kill() {
-          await chromium.close();
+      page = await chromium.newPage();
+    } catch (err) {
+      throw new Error(`Caught error spawning Chromium` + err);
+      return;
+    }
+
+    page.on('console', line => {
+      if (line._type === 'error') {
```

Comment:
Could we potentially extract this function into it's own utility and if possible flatten it, since 3 levels of nested `if` is quiet some nesting :-)

Also we are usually putting the closing `}` to the next line.